Aliucord/src/main/java/com/aliucord/coreplugins/NewPins.kt

```diff
@@ -0,0 +1,300 @@
+package com.aliucord.coreplugins
+
+import android.content.Context
+import android.view.View
+import android.widget.LinearLayout
+import android.widget.LinearLayout.LayoutParams
+import com.aliucord.Http
+import com.aliucord.Utils
+import com.aliucord.entities.CorePlugin
+import com.aliucord.patcher.*
+import com.aliucord.utils.GsonUtils
+import com.aliucord.utils.SerializedName
+import com.aliucord.utils.ViewUtils.addTo
+import com.discord.models.message.Message
+import com.discord.models.user.MeUser
+import com.discord.stores.*
+import com.discord.utilities.permissions.ManageMessageContext
+import com.discord.utilities.permissions.PermissionUtils
+import com.discord.utilities.rest.RestAPI
+import com.discord.views.TernaryCheckBox
+import com.discord.widgets.channels.permissions.WidgetChannelSettingsEditPermissions
+import com.discord.widgets.channels.permissions.`WidgetChannelSettingsEditPermissions$permissionCheckboxes$2`
+import com.discord.widgets.chat.list.adapter.WidgetChatListAdapter
+import com.discord.widgets.chat.list.entries.LoadingEntry
+import com.discord.widgets.chat.pins.WidgetChannelPinnedMessages
+import rx.Observable
+import java.net.URLEncoder
+import java.util.WeakHashMap
+
+internal class NewPins : CorePlugin(Manifest("NewPins")) {
+    override val isHidden: Boolean = true
+    override val isRequired: Boolean = true
+
+    companion object {
+        const val PIN_MESSAGES_PERMISSION = 1L shl 51
+    }
```

Comment:
This could be private I think
```kt
    private companion object {
        const val PIN_MESSAGES_PERMISSION = 1L shl 51
    }
```

Reply:
why's this even a companion can't this just be a top level const

Reply:
there's no difference really
top level consts are singleton classes anyways
and the const is inlined

Reply:
i think being a part of a companion better communicates ownership

Reply:
Yeah I put it in companion object so accessing it would require going through NewPins. I'll just make it private too.